engine/engine.go

```diff
@@ -948,3 +948,14 @@ func (bot *Engine) SetupExchanges() error {
 func (bot *Engine) WaitForInitialCurrencySync() error {
 	return bot.currencyPairSyncer.WaitForInitialSync()
 }
+
+// RegisterWebsocketDataHandler registers an externally defined data handler
+// for diverting and handling websocket notifications across all enabled
+// exchanges. InterceptorOnly as true will purge all other registered handlers
+// (including default) bypassing all other handling.
+func (bot *Engine) RegisterWebsocketDataHandler(fn WebsocketDataHandler, interceptorOnly bool) error {
```

Comment:
I am having a hard time conceptualizing the need for this because looking at it from an external perspective, if I have double registered, I would kill the program and fix the mistake, not hook into the engine and check if len(handlers) > n then set default handler via an exported function. Sorry if I am misunderstanding the idea.

Reply:
We're exporting a function which allows additional data handlers with no way of resetting it and has no way of validating passed in functions. If one is no longer needed, or something needs to change, the solution can't be to restart an application which manages people's orders with real money. We need a way of managing it, and I felt like resetting was the simplest solution out of the ones I've made